controller/src/beerocks/master/tasks/task.cpp

```diff
@@ -46,9 +46,7 @@ void task::response_received(std::string mac,
 void task::event_received(int event_type, void *obj)
 {
     auto it = pending_events.find(event_type);
-    if (it == pending_events.end()) {
-        TASK_LOG(DEBUG) << "received non-pending event " << event_type;
-    } else {
+    if (it != pending_events.end()) {
```

Comment:
I don't get this. Each task is registering for events, right? And in the controller we pass events only to tasks which should receive these events. If this is not the case, it is a bug, and this commit is just hiding it instead of fixing it.
Please correct me if i'm wrong..

Reply:
> Each task is registering for events, right?

No.
A task doesn't register to anything. A task can decide that it is pending for an event to block it work cycle until the event is being received.
It is normal behaviour that events would get to a task even if it is not pending for it.

Reply:
Well I think its not an ideal design choice, why is it implemented like that? Why can't a task "subscribe" to events? we can have the base class drop events for which the task didn't subscribe to. @vitalybu ?

Reply:
Why do you think that a task subscribing to an event is an ideal design? IMO it is not.
What is the benefit of such registration? Nothing.
It is not like when you push event, its reach to all the tasks and they have to filter it if they are subscribing to it or not. When we send an event we send it to a specific task ID, so only one instance of a specific task class would get it.
In the current design, each task has it is own event types, and on subscribing event design it would have to be global shared events for all the tasks, like TLVF, but that is just not necessary.